OpenChange/RTFHandler.m

```diff
@@ -378,26 +405,149 @@ - (RTFColorDef *) colorDefAtIndex: (unsigned int) theIndex
 //
 @implementation RTFHandler
 
-- (id) initWithData: (NSData *) theData
+static NSMapTable *_charsets = nil; 
+static NSMapTable *_cws = nil;
+typedef enum {
+  CW_UNKNOWN = 0,
+  CW_ANSICPG,
+  CW_B,
+  CW_CF,
+  CW_COLORTBL,
+  CW_F,
+  CW_FONTTBL,
+  CW_I,
+  CW_PAR,
+  CW_PICT,
+  CW_SOFTLINE,
+  CW_STRIKE,
+  CW_STYLESHEET,
+  CW_TAB,
+  CW_U,
+  CW_UL,
+  CW_ULNONE
+} CW_ID;
+
+static NSMapTable *_fontCws = nil;
+typedef enum {
+  FONTCW_UNKNOWN = 0,
+  FONTCW_F,
+  FONTCW_FBIDI,
+  FONTCW_FCHARSET,
+  FONTCW_FDECOR,
+  FONTCW_FMODERN,
+  FONTCW_FNIL,
+  FONTCW_FPRQ,
+  FONTCW_FROMAN,
+  FONTCW_FSCRIPT,
+  FONTCW_FSWISS,
+  FONTCW_FTECH
+} FONTCW_ID;
+
+static void _init_charsets_table()
 {
-  if ((self = [super init]))
-    {
-      ASSIGN(_data, theData);
-      _bytes = (char *)[_data bytes];
-      _len = [_data length];
-      _current_pos = 0;
-      
-      _charsets = NSCreateMapTable(NSObjectMapKeyCallBacks, NSNonOwnedPointerMapValueCallBacks, 10);
+      _charsets = NSCreateMapTable(NSObjectMapKeyCallBacks, NSNonOwnedPointerMapValueCallBacks, 23);
```

Comment:
Any reason to use this functions instead of NSDictionary and other classes? (check Codepages.m for similar code to this)